src/detect-engine-content-inspection.c

```diff
@@ -551,6 +553,15 @@ int DetectEngineContentInspection(DetectEngineCtx *de_ctx, DetectEngineThreadCtx
         SCLogDebug("lua match");
         goto match;
 #endif /* HAVE_LUA */
+    } else if (sm->type == DETECT_BASE64_DECODE) {
+        if (DetectBase64DecodeDoMatch(det_ctx, s, sm, buffer, buffer_len)) {
+            if (s->sm_arrays[DETECT_SM_LIST_BASE64_DATA] != NULL) {
+                KEYWORD_PROFILING_END(det_ctx, sm->type, 1);
+                if (DetectBase64DataDoMatch(de_ctx, det_ctx, s, f)) {
+                    goto match;
```

Comment:
The rule setup code should have made sure there is no SigMatch to check after this, right? Wonder if we should return directly, or jump to a new 'final_match' or similar.

Alternatively, a comment about this plus a DEBUG_VALIDATE_BUG_ON(sm->next != NULL) could also make it clear to anyone it's final.